src/libexpr/primops/fromYAML.cc

```
@@ -0,0 +1,156 @@
+
+            for (ryml::ConstNodeRef child : t.children()) {
+                std::string_view key(child.key().begin(), child.key().size());
+                visitYAMLNode(context, attrs.alloc(key), child);
+            }
+
+            v.mkAttrs(attrs);
+        }
+    } else if (t.is_seq()) {
+        context.state.mkList(v, t.num_children());
+
+        size_t i = 0;
+        for (ryml::ConstNodeRef child : t.children())
+            visitYAMLNode(context, *(v.listElems()[i++] = context.state.allocValue()), child);
+    } else if (valTypeCheck(ryml::TAG_NULL) && t.val_is_null()) {
+        v.mkNull();
+    } else if (t.has_val()) {
+        bool _bool;
+        NixFloat _float;
+        NixInt _int;
+        bool isQuoted = t.is_val_quoted();
+        auto val = t.val();
+        // Caution: ryml is able to convert integers into booleans and ryml::from_chars might ignore trailing chars
+        if (t.has_val_tag() && t.val_tag() == "!" && val.empty() && !isQuoted) { // special case for YAML string "!"
+            v.mkNull();
+        } else if (valTypeCheck(ryml::TAG_INT, !isQuoted) && val.is_integer() && ryml::from_chars(val, &_int)) {
+            v.mkInt(_int);
+        } else if (valTypeCheck(ryml::TAG_BOOL, !isQuoted) && ryml::from_chars(val, &_bool)) {
+            v.mkBool(_bool);
+        } else if (valTypeCheck(ryml::TAG_FLOAT, !isQuoted) && val.is_number() && ryml::from_chars(val, &_float)) {
+            v.mkFloat(_float);
+        }
+        if (valTypeCheck(ryml::TAG_STR) && v.type() == nThunk) {
+            std::string_view value(val.begin(), val.size());
+            v.mkString(value);
+        }
+    }
+    if (v.type() == nThunk) {
+        auto val = t.has_val() ? t.val() : "";
+        auto tag = t.has_val_tag() ? t.val_tag() : "";
+        auto msg = ryml::formatrs<std::string>(
+            "Error: The YAML value '{}' with '{}' tag cannot be represented as Nix data type", val, tag);
```

The test suite should have some tests for YAML that cannot be represented (i.e. asserting that we get an error message).

---

With the latest commit inputs, which shall throw an error, are tested, too, although some tests are skipped because rapidyaml still parses these inputs. I think that these cases could be fixed upstream in rapidyaml over the next months. At least I can spent some time on it.